src/wallet/spend.h

```diff
@@ -55,23 +55,34 @@ struct CoinsResult {
     CAmount total_amount{0};
 };
 
+struct CoinFilterParams {
+    // Outputs below the minimum amount will not get selected
+    CAmount min_amount{1};
+    // Outputs above the maximum amount will not get selected
+    CAmount max_amount{MAX_MONEY};
+    // Return outputs until the minimum sum amount is covered
+    CAmount min_sum_amount{MAX_MONEY};
+    // Maximum number of outputs that can be returned
+    uint64_t max_count{0};
+    // By default, return only spendable outputs
+    bool only_spendable{true};
+    // By default, do not include immature coinbase outputs
+    bool include_immature_coinbase{false};
+};
+
 /**
  * Populate the CoinsResult struct with vectors of available COutputs, organized by OutputType.
  */
 CoinsResult AvailableCoins(const CWallet& wallet,
                            const CCoinControl* coinControl = nullptr,
                            std::optional<CFeeRate> feerate = std::nullopt,
-                           const CAmount& nMinimumAmount = 1,
-                           const CAmount& nMaximumAmount = MAX_MONEY,
-                           const CAmount& nMinimumSumAmount = MAX_MONEY,
-                           const uint64_t nMaximumCount = 0,
-                           bool only_spendable = true) EXCLUSIVE_LOCKS_REQUIRED(wallet.cs_wallet);
+                           const CoinFilterParams& params = {}) EXCLUSIVE_LOCKS_REQUIRED(wallet.cs_wallet);
 
 /**
- * Wrapper function for AvailableCoins which skips the `feerate` parameter. Use this function
+ * Wrapper function for AvailableCoins which skips the `feerate` and `CoinFilterParams::only_spendable` parameters. Use this function
```

Comment:
Sorry, I'm not sure I follow here 😅 

Why are you saying that this function "skips" `CoinFilterParams::only_spendable`? To me, it seems that it's skipping only the `feerate` parameter (i.e., `AvailableCoins` takes a `feerate` param, and `AvailableCoinsListUnspent` doesn't)

Reply:
Internally, `AvailableCoinsListUnspent` sets `only_spendable` to false (this isn't a new behavior introduced on this PR, `AvailableCoinsListUnspent` callers expect to receive all the coins). Check one of my previous comments https://github.com/bitcoin/bitcoin/pull/25730#discussion_r994815499

So, what the comment basically says is "do not customize `CoinFilterParams::only_spendable`, it will not be taken into account".

Reply:
Ah, now it makes sense, thanks!